src/main/java/dinkplugin/notifiers/KillCountNotifier.java

```diff
@@ -0,0 +1,146 @@
+package dinkplugin.notifiers;
+
+import dinkplugin.DinkPlugin;
+import dinkplugin.Utils;
+import dinkplugin.message.NotificationBody;
+import dinkplugin.message.NotificationType;
+import dinkplugin.notifiers.data.BossNotificationData;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.NPC;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.tuple.Pair;
+import org.jetbrains.annotations.Nullable;
+
+import javax.inject.Inject;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@Slf4j
+public class KillCountNotifier extends BaseNotifier {
+    private static final Pattern PRIMARY_REGEX = Pattern.compile("Your (?<key>.+)\\s(?<type>kill|chest|completion)\\s?count is: (?<value>\\d+)\\b");
+    private static final Pattern SECONDARY_REGEX = Pattern.compile("Your (?:completed|subdued) (?<key>.+) count is: (?<value>\\d+)\\b");
+
+    @Inject
+    public KillCountNotifier(DinkPlugin plugin) {
+        super(plugin);
+    }
+
+    @Override
+    public boolean isEnabled() {
+        return plugin.getConfig().notifyKillCount() && super.isEnabled();
+    }
+
+    public void onGameMessage(String message) {
+        if (isEnabled())
+            parse(message).ifPresent(pair -> handleKill(pair.getLeft(), pair.getRight(), message));
+    }
+
+    private void handleKill(String boss, int killCount, String rawMessage) {
+        if (!checkKillInterval(killCount)) return;
+
+        // Assemble content
+        String player = Utils.getPlayerName(plugin.getClient());
+        String content = StringUtils.replaceEach(
+            plugin.getConfig().killCountMessage(),
+            new String[] { "%USERNAME%", "%BOSS%", "%COUNT%" },
+            new String[] { player, boss, String.valueOf(killCount) }
+        );
+
+        // Prepare body
+        NotificationBody.NotificationBodyBuilder<BossNotificationData> body =
+            NotificationBody.<BossNotificationData>builder()
+                .content(content)
+                .extra(new BossNotificationData(boss, killCount, rawMessage))
+                .playerName(player)
+                .type(NotificationType.KILL_COUNT);
+
+        // Add embed if not screenshotting
+        boolean screenshot = plugin.getConfig().killCountSendImage();
+        if (!screenshot)
+            Arrays.stream(plugin.getClient().getCachedNPCs())
```

Comment:
This approach doesn't work for Wintertodt, Barrows, or the raids. For Wintertodt, I can hardcode the howling storm gif, if desired.